### DISCUSSION TITLE
simplifying-dependency-injection-for-iconfigureoptions-with-the-configureoptions-helper/

### DISCUSSION THREAD

**Discussion by giscus** [andrewlock/blog-comments#94] [Announcements]
# simplifying-dependency-injection-for-iconfigureoptions-with-the-configureoptions-helper/

In this post I show how to use ConfigureOptions() to automatically register all the IConfigureOptions interfaces implemented by a type with the DI container

https://andrewlock.net/simplifying-dependency-injection-for-iconfigureoptions-with-the-configureoptions-helper/

[a comment by an automated account is not shown]

**Reply by andrewlock** [reply]
<em>[Sock](https://disqus.com/by/andrewlockdotnet/) (@andrewlock) commented [on Disqus](https://disqus.com/home/forum/andrewlock/) at November 14 2018, 07:53, in reply to [graemechristie](https://disqus.com/by/graemechristie/)'s [comment](https://github.com/andrewlock/blog-comments/discussions/94#discussioncomment-6054160)</em>

---

<p>Oops, thanks, fixed</p>

[a comment by an automated account is not shown]

**Comment by ProbablePrime** [comment]
Needed this post, several months ago. Happy I found it today, thank you!

You have a typo in: `public class ConfigurePublicMySettingsOptions : IConfigureNmedOptions<MySettings>` 

Otherwise exactly what I needed, thank you!!!
